### ISSUE TITLE
Wake on Lan

### ISSUE THREAD

**Issue by NicLogix** [AdguardTeam/AdGuardHome#2888]
I would like the ability with DHCP to Wake on Lan with "magic packet", as the mac address is already listed when using Adguard Home as DHCP Server for network.

**Comment by ameshkov** [comment]
Uh, never heard of anything like that. Could you please provide some examples? Maybe some other software already has a similar feature?

**Comment by NicLogix** [comment]
As stated by HorayNarea, Wake on Lan is able to boot or bring pc out of suspend.  The "magic packet" is the most popular way of doing it.
This packet is sent on the broadcast address (ex. Subnet 192.168.0.0, sent on 192.168.0.255) on port 9.  It is a L2 - L3 routing protocol, so needs an ARP binding (Mac to IP).  This is usually done on the router.
The info to send the packet is already in Adguard when you set an IP reservation via Mac.
I do not know enough code to explain how to  implement this, but a separate page in the Web UI which can list the Adguard client, and send the packet would be full marks. A backend like etherwake might do the trick.

I only have Linux Debian PC's, but the concept is the same for Windoze.

**Comment by ameshkov** [comment]
Seems a little bit out of scope for us, but I'll mark it as feature request and we'll see if there are more people requesting it.

**Comment by hapylestat** [comment]
I not able to see any patter of WoL usage, or barely one. 

Back to OP:
 DHCP server is not an initiator of the connection, it only broadcast himself as present in the network with contact info (e.g. call me if you need). And after client initiate the DHCP workflow sequence. 

The possibility:
 AdGuard to keep list of known clients, to allow for user to wake them up so if the client is configured properly, it would receive the MagicPacket, boot up and would initiate DHCP client.

Pros: 
- ability to use AdGuard as computer management software? 

Cons:
- No guarantee that remote PC will boot, or would initiate DHCP client (miss-configuration, system boot error)
- It's not popular feature and is barely used
- Not really a part of DHCP+DNS ecosystem and the whole product idea


TL;DR;
Instead, AdGuard probably could implement a way to create a "Custom Actions" for the users, which could be added to menu/context menu.  So user would be able to integrate his own ideas, by creating his own Custom action, and AdGuard would supply required information like IP/MAC ...whatever.

**Comment by samsam-rolon** [comment]
Maybe it can be easily add with this work ?
https://github.com/sabhiram/go-wol
I use it but a "All-in-one" interface through the DHCP of AdGuard should be better

**Comment by nerocide** [comment]
This WOL feature would be great, especially if additionnaly exposed in api `/control` server

**Comment by levi2m** [comment]
this would be a fantastic feature, more one vote for this!

**Comment by tvup** [comment]
Different approach to the same need: [idle-less](https://github.com/tvup/idle-less) handles WoL at the reverse proxy layer. When HTTP/HTTPS traffic arrives and the backend is sleeping, it sends the magic packet automatically.

Works well alongside AdGuard Home — AdGuard handles DNS, idle-less handles the "wake on web request" part.

https://github.com/tvup/idle-less
